src/test/java/use_cases/add_edit_category_use_case/CategoryUCITest.java

```diff
@@ -0,0 +1,214 @@
+package use_cases.add_edit_category_use_case;
+
+import entities.EntityException;
+import entities.MonthlyStorage;
+import entities.SessionStorage;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import views.add_edit_category_views.CategoryP;
+
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * CategoryUCITest contains tests for the Category use cases (add & edit category).
+ */
+class CategoryUCITest {
+    private static SessionStorage session;
+
+    @BeforeAll
+    public static void categorySetUp() {
+        /*
+          Creates a SessionStorage for the following test cases.
+         */
+        session = new SessionStorage();
+    }
+
+    @Test
+    void addCategoryInMonthSuccess() throws EntityException {
+        /*
+          Tests success add case by checking if the size of categoryData in MonthlyStorage is correctly updated.
+          Adding two known-to-fail inputs and two successful inputs.
+         */
```

Comment:
This Javadoc should go above the test method. Also, your input data inputting a variable null that you won't get, which made you pass a failing test.

Reply:
That null is not a user input. If you check AddCategoryV class, the null is an assignment. It is just so the CategoryC can call both types of use cases with the same method CategoryC.categoryInMonth(). (This is an accommodation to what Ari suggested about combining the input data!) 

`public AddCategoryV(CategoryC controller,int monthID, SessionStorage currSession)`: this is the constructor of AddCategoryV so don't worry.

Reply:
I see! in that case, the same problem mentioned in input data class should also be OK. I'll mark that as resolved too!